recipes/install.rb

```diff
@@ -848,6 +857,24 @@
   action :create
 end
 
+if conda_helpers.is_upgrade && Gem::Version.new(node['install']['current_version']) < Gem::Version.new('3.1.0')
+  # We need to restore the internal certificates and the flyway schemas for the upgrade
+  # To be successfull
+  bash 'Move old certificates and flyway schemas' do
+    user 'root'
+    code <<-EOH
+      set -e
+      cp -p #{node['hopsworks']['domains_dir']}4/domain1/config/internal* #{node['hopsworks']['config_dir']}/
+      cp -p #{node['hopsworks']['config_dir']}/internal_bundle.crt #{node['hopsworks']['config_dir']}/certificate_bundle.pem
+
+      # Hardcoded because the attribute doesn't exists anymore
+      cp #{node['hopsworks']['dir']}/certs-dir/certs/ca.cert.pem #{node['hopsworks']['config_dir']}/root_ca.pem
+
+      cp -p #{node['hopsworks']['domains_dir']}4/flyway/sql/* #{theDomain}/flyway/sql/
```

Comment:
And also `chown #{node['hopsworks']['user']} /srv/hops/certs-dir/certs/ca.cert.pem`